### PR TITLE
remove Verifiable Credentials and Presentations from store

### PR DESCRIPTION
**Pull request by arttor**
**Add methods to remove VC and VP from the store**

Currently, there is no way to remove issued VC and VP from the store so I've added required methods

### CODE REVIEW

**Comment by troyronda** [comment]
@arttor Thanks for your contribution. Please note that Hyperledger repositories require DCO sign-off. See [failing check](https://github.com/hyperledger/aries-framework-go/pull/2028/checks?check_run_id=899390945).

File: pkg/store/verifiable/store.go [hunk 1]
```diff
@@ -43,6 +43,8 @@ type Store interface {
 	GetPresentationIDByName(name string) (string, error)
 	GetCredentials() ([]*Record, error)
 	GetPresentations() ([]*Record, error)
+	RemoveCredential(name string) error
+	RemovePresentation(name string) error
```
Comment (soluchok) [inline pkg/store/verifiable/store.go:47]: function's names should be `RemoveCredentialByName` and `RemovePresentationByName`

File: pkg/store/verifiable/store.go [hunk 2]
```diff
@@ -229,6 +231,62 @@ func (s *StoreImplementation) GetPresentations() ([]*Record, error) {
...
+// RemoveCredential removes the verifiable credential and its records containing given name.
+func (s *StoreImplementation) RemoveCredential(name string) error {
+	if name == "" {
+		return errors.New("credential name is mandatory")
+	}
+
+	id, err := s.GetCredentialIDByName(name)
+	if err != nil && !errors.Is(err, storage.ErrDataNotFound) {
+		return fmt.Errorf("get credential id using name : %w", err)
+	}
```
Comment (soluchok) [inline pkg/store/verifiable/store.go:243]: why do we need this condition  `!errors.Is(err, storage.ErrDataNotFound)` ?

File: pkg/store/verifiable/store.go [hunk 3]
```diff
@@ -229,6 +231,62 @@ func (s *StoreImplementation) GetPresentations() ([]*Record, error) {
...
+}
+
+// RemovePresentation removes the verifiable presentation and its records containing given name.
+func (s *StoreImplementation) RemovePresentation(name string) error {
+	if name == "" {
+		return errors.New("presentation name is mandatory")
+	}
+
+	id, err := s.GetPresentationIDByName(name)
+	if err != nil && !errors.Is(err, storage.ErrDataNotFound) {
```
Comment (soluchok) [inline pkg/store/verifiable/store.go:260]: same here `!errors.Is(err, storage.ErrDataNotFound)` ?

File: pkg/store/verifiable/store.go (hunk 1 above)
Comment (arttor) [reply pkg/store/verifiable/store.go]: Done

File: pkg/store/verifiable/store.go (hunk 2 above)
Comment (arttor) [reply pkg/store/verifiable/store.go]: Correct. we don't need this check here and it is ok to rerun this error message for missing ID too.
Fixed.

File: pkg/store/verifiable/store.go (hunk 3 above)
Comment (arttor) [reply pkg/store/verifiable/store.go]: Fixed

File: pkg/store/verifiable/store.go [hunk 4]
```diff
@@ -229,6 +231,62 @@ func (s *StoreImplementation) GetPresentations() ([]*Record, error) {
...
+	}
+
+	return nil
+}
+
+func (s *StoreImplementation) remove(id, recordKey string) error {
+	if id == "" {
+		return errors.New("id is required")
+	}
+
```
Comment (soluchok) [inline pkg/store/verifiable/store.go:276]: nit: It will never happen, I would remove it.

Comment (arttor) [reply pkg/store/verifiable/store.go]: Done.

Comment (arttor) [reply pkg/store/verifiable/store.go]: Can you please merge PR if everything is ok?

Comment (soluchok) [reply pkg/store/verifiable/store.go]: I can't :) @troyronda @fqutishat @llorllale ^

**Pull request merged**